### PR TITLE
stm32: dma: Simplify handling of Kconfig symbols

### PR DESCRIPTION
**Pull request by erwango**
In order to simplify the handling of DMA_STM32_V1/V2 and DMAMUX_STM32
symbols, set them directly based on related compatible status.

As a preliminary step, instantiate matching compatibles.

### CODE REVIEW

**Comment by erwango** [comment]
^^@shlomow

File: dts/bindings/dma/st,stm32-dma-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     dma2: dma-controller@40020400 {
+         compatible = "st,stm32-dma-v2";
+         ...
+         st,mem2mem;
+         dma-requests = <7>;
+         status = "disabled";
+         label = "DMA_2";
+        };
+
+  For the client part, example for stm32l47 on DMA1 instance
```
Comment (tagunil) [inline dts/bindings/dma/st,stm32-dma-v2.yaml:60]: l476?

File: dts/bindings/dma/st,stm32-dma-v2.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+properties:
+    reg:
+      required: true
+
+    interrupts:
+      required: true
+
+    st,mem2mem:
+      type: boolean
+      description: If the DMA controller V1 supports memory to memory transfer
```
Comment (tagunil) [inline dts/bindings/dma/st,stm32-dma-v2.yaml:83]: V2?

File: drivers/dma/Kconfig.stm32 [hunk 3]
```diff
@@ -15,20 +19,20 @@ if DMA_STM32
 
 config DMA_STM32_V1
 	bool
-	depends on SOC_SERIES_STM32F2X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
+	default $(DT_COMPAT_ST_STM32_DMA_V1)
 	help
```
Comment (shlomow) [inline drivers/dma/Kconfig.stm32:23]: I think the help strings should be changed too

File: dts/arm/st/h7/stm32h7.dtsi [hunk 4]
```diff
@@ -678,7 +678,7 @@
 		};
 
 		dma2: dma@40020400 {
-			compatible = "st,stm32-dma";
+			compatible = "st,stm32-dma-v2";
 			#dma-cells = <4>;
 			reg = <0x40020400 0x400>;
 			interrupts = <11 0>, <12 0>, <13 0>, <14 0>, <15 0>, <16 0>,
```
Comment (shlomow) [inline dts/arm/st/h7/stm32h7.dtsi:684]: not really related to this, but the `interrupts` property should be `interrupts = <56 0>, <57 0>, <58 0>, <59 0>, <60 0>, <68 0>, <69 0>, <70 0>;`

**Review by FRASTM — APPROVED** [review]

File: soc/arm/st_stm32/stm32l4/Kconfig.defconfig.series [hunk 5]
```diff
@@ -16,10 +16,5 @@ config SOC_SERIES
 config STM32_LPTIM_TIMER
 	bool
 	default y
-	depends on PM
```
Comment (FRASTM) [inline soc/arm/st_stm32/stm32l4/Kconfig.defconfig.series:19]: why removing "depends on PM" there

**Review by FRASTM — APPROVED** [review]
else LGTM

File: soc/arm/st_stm32/stm32l4/Kconfig.defconfig.series (hunk 5 above)
Comment (erwango) [reply soc/arm/st_stm32/stm32l4/Kconfig.defconfig.series:19]: Not related indeed. Don't know how this ended up here. Will fix

File: dts/bindings/dma/st,stm32-dma-v2.yaml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+       	       0x2: Word (32 bits)
+       	       0x3: reserved
+        -bit 15: Reserved
+        -bit 16-17 : Priority level
+       	       0x0: low
+       	       0x1: medium
+       	       0x2: high
+       	       0x3: very high
+    4. features: A 32bit bitfield value specifying DMA features
+        -bit 0-1: DMA FIFO threshold selection (not USED for DMA V2)
```
Comment (erwango) [inline dts/bindings/dma/st,stm32-dma-v2.yaml:45]: @FRASTM, I'm not sure about this part.
Does it mean that "features" has no use for V2 ?
(Even if not used, it should be kept for to simplify things at users level, but then I need to make it clear).

File: dts/bindings/dma/st,stm32-dma-v2.yaml (hunk 1 above)
Comment (erwango) [reply dts/bindings/dma/st,stm32-dma-v2.yaml]: Done, txs for the catch

File: drivers/dma/Kconfig.stm32 (hunk 3 above)
Comment (erwango) [reply drivers/dma/Kconfig.stm32:23]: Indeed, done

File: dts/arm/st/h7/stm32h7.dtsi (hunk 4 above)
Comment (erwango) [reply dts/arm/st/h7/stm32h7.dtsi]: Thx for catching that. I've fixed this as well.

File: dts/bindings/dma/st,stm32-dma-v2.yaml (hunk 6 above)
Comment (FRASTM) [reply dts/bindings/dma/st,stm32-dma-v2.yaml:45]: yes fifo threshold is not used in case of DMA V2

Comment (erwango) [reply dts/bindings/dma/st,stm32-dma-v2.yaml]: ok, txs. I've made it clear in the binding.

File: dts/arm/st/h7/stm32h7.dtsi [hunk 7]
```diff
@@ -667,7 +667,7 @@
 		};
 
 		dma1: dma@40020000 {
-			compatible = "st,stm32-dma";
+			compatible = "st,stm32-dma-v2";
```
Comment (shlomow) [inline dts/arm/st/h7/stm32h7.dtsi:670]: h7 is v1, no?

Comment (erwango) [reply dts/arm/st/h7/stm32h7.dtsi:670]: Good catch, will fix

Comment (erwango) [reply dts/arm/st/h7/stm32h7.dtsi]: Done

File: drivers/dma/Kconfig.stm32 [hunk 8]
```diff
@@ -4,6 +4,10 @@
 # Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
 # SPDX-License-Identifier: Apache-2.0
 
+DT_COMPAT_ST_STM32_DMA_V1 := $(dt_compat_enabled,$(st,stm32-dma-v1))
+DT_COMPAT_ST_STM32_DMA_V2 := $(dt_compat_enabled,$(st,stm32-dma-v2))
+DT_COMPAT_ST_STM32_DMAMUX := $(dt_compat_enabled,$(st,stm32-dmamux))
```
Comment (mbolivar-nordic) [inline drivers/dma/Kconfig.stm32:9]: The `$(st,stm32-dma-v1)` part doesn't look right to me. There's no kconfigfunction named `st`.

Are you sure about this?

Comment (erwango) [reply drivers/dma/Kconfig.stm32]: Late untested fix. Txs !

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**